src/yarp.c

```diff
@@ -1016,6 +1016,62 @@ yp_block_argument_node_create(yp_parser_t *parser, const yp_token_t *operator, y
     return node;
 }
 
+static void
+YP_ATTRIBUTE_UNUSED yp_scope_node_create(yp_parameters_node_t *parameters, yp_statements_node_t *statements, yp_constant_id_list_t locals, const char *start, const char *end, yp_scope_node_t *dest)
+{
+    *dest = (yp_scope_node_t) {
+         {
+             .type = YP_NODE_SCOPE_NODE,
+             .location = { .start = start, .end = end },
+         },
+        .parameters = parameters,
+        .statements = statements,
+        .locals = locals,
+    };
+    return;
+}
+
+// TODO: Implement this for all other nodes which can have a scope
+void
+yp_get_scope_node(yp_node_t *node, yp_scope_node_t *dest) {
```

Comment:
Just in terms of naming, the codebase has loosely been `yp_` + noun + verb. I haven't written that down anywhere though, so that's on me! But would you be able to rename this to `yp_scope_node_init` or something similar?